build/golang_test.go

```diff
@@ -48,3 +50,16 @@ func validateModule(t *testing.T, module buildinfo.Module, expectedDependencies,
 		assert.NotEmpty(t, module.Dependencies[0].Checksum.Sha256, "Empty SHA256 field.")
 	}
 }
+
+func validateRequestedBy(t *testing.T, module entities.Module) {
+	for _, dep := range module.Dependencies {
+		if assert.NotEmpty(t, dep.RequestedBy) {
+			if assert.NotEmpty(t, dep.RequestedBy[0]) {
+				if dep.Id == "github.com/pkg/errors:v0.8.0" {
+					assert.Len(t, dep.RequestedBy, 2, "'errors:v0.8.0' dependency should contain 2 requestedBy paths")
+				}
+			}
+		}
```

Comment:
I'd prefer using this attitude:
```suggestion
		switch dep.Id {
		case "github.com/pkg/errors:v0.8.0":
			assert.Equal(t, [][]string{{"github.com/jfrog/gofrog:v1.1.1", "github.com/jfrog/dependency"}, {"github.com/jfrog/dependency"}}, dep.RequestedBy)
		case "golang.org/x/text:v0.0.0-20170915032832-14c0d48ead0c":
			assert.Equal(t, [][]string{{"rsc.io/sampler:v1.3.0", "rsc.io/quote:v1.5.2", "github.com/jfrog/dependency"}}, dep.RequestedBy)
		...
		default:
			assert.Fail(t, "Unexpected dependency "+dep.Id)
		}
```